pkg/collector/corechecks/snmp/devicecheck/devicecheck.go

```diff
@@ -74,11 +75,19 @@ func (d *DeviceCheck) GetIDTags() []string {
 }
 
 // GetDeviceHostname returns DeviceID as hostname if UseDeviceIDAsHostname is true
-func (d *DeviceCheck) GetDeviceHostname() string {
+func (d *DeviceCheck) GetDeviceHostname() (string, error) {
 	if d.config.UseDeviceIDAsHostname {
-		return deviceHostnamePrefix + d.config.DeviceID
+		hostname := deviceHostnamePrefix + d.config.DeviceID
+		normalizedHostname := util.NormalizeHost(hostname)
```

Comment:
Now that we have `ValidateNamespace`, I'm wondering if there is a need for `util.NormalizeHost` at all (change to GetDeviceHostname()). 

I feel it's adding more complexity to code for a case that seems to never happen and even if it happens, it's not critical anymore (namespace is already validated).

Reply:
Let's discuss about it over zoom !